src/linker/Linker/SignatureParser.cs

```diff
@@ -0,0 +1,661 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using Mono.Cecil;
+using Mono.Collections.Generic;
+
+namespace Mono.Linker
+{
+	/// <summary>
+	///  Parses a signature for a member, in the format used for C# Documentation Comments:
+	///  https://github.com/dotnet/csharplang/blob/master/spec/documentation-comments.md#id-string-format
+	///  Adapted from Roslyn's DocumentationCommentId:
+	///  https://github.com/dotnet/roslyn/blob/master/src/Compilers/Core/Portable/DocumentationCommentId.cs
+	/// </summary>
+	///
+	/// Roslyn's API works with ISymbol, which represents a symbol exposed by the compiler.
+	/// a Symbol has information about the source language, name, metadata name,
+	/// containing scopes, visibility/accessibility, attributes, etc.
+	/// This API instead works with the Cecil OM. It can be used to refer to IL definitions
+	/// where the signature of a member can contain references to instantiated generics.
+	///
+	public static class SignatureParser
```

Comment:
It'd be better to use a less ambiguous name (there are a lot of signatures inside linker)

Reply:
Changed to DocumentationSignatureParser